mrmustard/physics/bargmann.py

```diff
@@ -94,14 +91,14 @@ def wigner_to_bargmann_Choi(X, Y, d):
         [[I, 1j * I, o, o], [o, o, I, -1j * I], [I, -1j * I, o, o], [o, o, I, 1j * I]]
     ) / np.sqrt(2)
     A = math.matmul(math.matmul(R, A), math.dagger(R))
-    A = math.matmul(A, math.Xmat(2 * N))  # yes: X on the right
+    A = math.matmul(math.Xmat(2 * N), A)
     b = math.matvec(xi_inv, d)
     B = math.matvec(math.conj(R), math.concat([b, -math.matvec(XT, b)], axis=-1)) / math.sqrt(
         settings.HBAR, dtype=R.dtype
     )
-    B = math.concat([B[2 * N :], B[: 2 * N]], axis=-1)  # yes: opposite order
+    B = math.asnumpy(B)
     C = math.exp(-0.5 * math.sum(d * b) / settings.HBAR) / math.sqrt(math.det(xi), dtype=b.dtype)
-    # now A and B have order [out_l, in_l out_r, in_r].
+    # now A and B have order [out_r, in_r out_l, in_l].
```

Comment:
we want [out_l, in_l, out_r, in_r] no?

Reply:
as I understand, we need [out_l, in_l out_r, in_r] before the Fock tensor, but here for ABC the order is [out_r, in_r out_l, in_l].

Reply:
if you find any bugs, feel free to fix them as part of this PR (and add a test that would have failed)